### DISCUSSION TITLE
Address connected through Fireblocks always shows as zero address

### DISCUSSION THREAD

**Discussion by mitch-PV01** [WalletConnect/walletconnect-monorepo#3955] [Developer Support] [answered]
Hi,

I am using web3modal v2.10.2, when selecting Fireblocks, and selecting an address in the fireblocks console, the wallet connect button shows as being connected to the Ethereum zero address (`0x0000000000000000000000000000000000000000`). I have tried numerous times in both firefox and chrome.

![Screenshot from 2023-11-20 17-17-09](https://github.com/WalletConnect/walletconnect-monorepo/assets/122266026/22eadbbe-f2fd-4d9e-ad1f-866a98cd6d76)

I was able to replicate the issue for web3modal v3 in the demo [here](https://web3modal-laboratory-v3.vercel.app/library/wagmi/), which demonstrates it's not an issue with the calling code:

![Screenshot from 2023-11-20 17-20-36](https://github.com/WalletConnect/walletconnect-monorepo/assets/122266026/38768f11-b0bc-4e3f-8cda-0464424a64e4)

![Screenshot from 2023-11-20 17-15-59](https://github.com/WalletConnect/walletconnect-monorepo/assets/122266026/ce81d153-f717-4685-a487-9313bae19e2a)

I am not sure whether this is an issue with web3modal or with fireblocks?

Many thanks.

**Comment by glitch-txs** [comment]
Hi, could you share the steps to reproduce this issue?

**Reply by mitch-PV01** [reply]
Hi, please see video below. The steps are the following:
1. Go to https://web3modal-laboratory-v3.vercel.app/library/wagmi/
2. Click 'Connect Wallet'
3. Select 'Fireblocks'
4. Switch to 'WEBAPP'
5. Click 'OPEN', which opens the fireblocks console in another tab
6. Choose the address to connect
7. Click 'Connect'
8. Go back to web3modal demo
9. See that it is connected to zero eth address

[Screencast from 2023-11-21 11-16-21.webm](https://github.com/WalletConnect/walletconnect-monorepo/assets/122266026/d97d339f-55b3-41b3-959f-3e4476962b26)

**Reply by glitch-txs** [reply, marked as answer]
it's gonna be fixed by next month, this is an internal workaround for them that should only happen in testnet.

**Reply by mitch-PV01** [reply]
Ah, thanks for this. Good to know this should work on mainnet. It will be useful once it's working on testnet as well :+1:
